listener/listener.py

```diff
@@ -0,0 +1,234 @@
+                                        status_msg="Skipped advisor result due to missing insights entitlement.")
+            LOGGER.debug("Skipped advisor result due to missing insights entitlement.")
+            return False
+        return True
+
+
+class Listener:
+    """Main listener class"""
+
+    def __init__(self, db_pool: Pool, loop: asyncio.BaseEventLoop):
+        """Constructor"""
+        self.consumer = MQReader([CFG.listener_ingress_topic])
+        self.payload_tracker = MQWriter([CFG.payload_tracker_topic])
+
+        self.inventory_msg_validator = InventoryMsgValidator(self.payload_tracker)
+        self.advisor_msg_validator = AdvisorMsgValidator(self.payload_tracker)
+
+        self.loop = loop
+
+        self.queue = ListenerQueue(db_pool, self.payload_tracker, loop)
+        self._queue_task = None
+
+    async def init(self):
+        """Async constructor"""
+        await self.queue.init()
+
+    def _consume_inventory_msg(self, msg: dict) -> InventoryMsgType:
+        """Consumes inventory message"""
+        is_ok, msg_type = self.inventory_msg_validator.validate_message(msg)
+        if not is_ok:
+            LOGGER.error("obtained invalid inventory msg")
+            return InventoryMsgType.UNKNOWN
+        if msg_type is InventoryMsgType.UPLOAD:
+            self.queue.push_inventory_msg(msg["host"]["org_id"], msg["host"]["id"], msg, msg_type)
+        elif msg_type is InventoryMsgType.DELETE:
+            self.queue.push_inventory_msg(msg["org_id"], msg["id"], msg, msg_type)
+        return msg_type
+
+    def _consume_advisor_msg(self, msg: dict):
+        """Consumes advisor message"""
+        is_ok = self.advisor_msg_validator.validate_message(msg)
+        if not is_ok:
+            LOGGER.error("obtained invalid advisor msg")
+            return
+        self.queue.push_advisor_msg(msg["input"]["host"]["org_id"], msg["input"]["host"]["id"], msg)
+
+    async def stop(self):
+        """Stops whole listener process"""
+        await self.consumer.stop()
+        await self.payload_tracker.stop()
+        await self.queue.stop()
+
+    async def run(self):
+        """Stars the listener service"""
+        await self.consumer.start()
+
+        try:
+            async for msg in self.consumer.client:
+                self.loop.create_task(self.consume_message(msg))
+        finally:
+            await self.stop()
+
+    async def consume_message(self, msg: ConsumerRecord):
+        """Consumes single message from advisor or inventory"""
+        try:
+            msg_dict = json.loads(msg.value.decode("utf-8"))
+        except json.decoder.JSONDecodeError as exc:
+            MESSAGE_PARSE_ERROR.inc()
+            LOGGER.exception("Unable to parse message: %s", exc)
+            return
+
+        if msg_dict.get("host", None) is not None or msg_dict.get("type", None) == "delete":
+            self._consume_inventory_msg(msg_dict)
+        elif msg_dict.get("input", None) is not None:
```

Comment:
```suggestion
        if msg_dict.get("host") is not None or msg_dict.get("type") == "delete":
            self._consume_inventory_msg(msg_dict)
        elif msg_dict.get("input") is not None:
```
dict.get() default is None
I don't know what is the value for "input", or "host", maybe it is not even needed to do `is not None` if you don't need to care about e.g. empty string

Reply:
edited it everywhere